guides/security.md

```diff
@@ -0,0 +1,65 @@
+
+Since an offline computer has no connection, it is extremely hard for someone without physical access to it to access the account’s keys. However, this also makes every transaction an extremely manual process. A common practice instead is to maintain two accounts: one offline account that securely holds the majority of your assets and another online account that holds only a few assets. Most transactions can be performed with the online account and, when its funds are low, a person can manually replenish it from the offline account.
+
+You can think of this method like having both a bank vault and a cash register drawer. Most of the time the vault is closed and locked. It is only opened occasionally (and under specific procedures) to replenish a register drawer that is running low or to store excess funds from a register drawer that is overflowing. If someone attempts to rob the bank, it is extremely hard for them to get away with anything more than what was in the register drawer.
+
+
+## Require multiple authorizations or signers
+
+Sensitive accounts can be secured by requiring authorization from multiple individuals to make a transaction. Read the [multisignature guide](concepts/multi-sig.md) to learn more about how.
+
+If you require multiple signers, you should also ensure that you do not require all the possible signers to sign a transaction. If one of the signers loses the keys to their account, you will no longer be able to perform transactions if they have to sign them.
+
+
+## Ensure assets are revocable
+
+If you issue your own assets, you should usually ensure that they can be revoked using the [“authorization revocable” flag on the account](concepts/accounts.md#flags). This allows you to effectively freeze your assets in someone else’s account in case of theft or in other extenuating circumstances.
+
+
+## What if an account’s keys are compromised?
+
+Because Stellar’s security is based around public key encryption, it’s critical that an account’s secret seed is not shared. Anyone who has access to the seed effectively has control of the account. However, if someone learns your account’s seed or you accidentally share it with someone who shouldn’t know it, you can remove its ability to control the account with the following steps:
+
+1. Make a new account.
+2. Add the new account as a signer on the compromised account. (Use the [`set options` operation](concepts/list-of-operations.md#set-options).
+3. Remove the compromised key’s signing authority on the compromised account.
+4. Now the new account controls the compromised account and the compromised account’s keys are no longer able to sign transactions.
+
+(NOTE: You must notify the owners of other accounts that the compromised account has signing authority or that the keys are compromised. They need to remove the compromised account as a signer as well [and add the new account].)
```

Comment:
(Discussion from Google Docs…)

@mr0grog: Is this accurate? Or is signing transitive in this way—if account A requires signing from account B (the compromised account), which cannot sign for itself but instead requires signing from account C, does A require signing from C?

(This is prompted by the third paragraph here, which is not quite clear to me in this regard: https://www.stellar.org/developers/learn/concepts/multi-sig.html#transaction-signatures)

@bartekn: Not sure if I understand the question correctly so if it doesn't make sense let me know:

There are multiple procedures on how signatures can be added:

1. Signers can sign a transaction and send envelope to the "next" signer, after adding the last required signature tx can be submitted to the network.
2. Each signer can add a signature and send an envelope to a "central server" that will merge all envelopes and generate a single envelope with all signatures.

Answering your question:
For the method 1) signers should be informed about invalidated keys (or simply should check the current list of signers using Horizon) because some malicious user can sign any transaction and, by sending it to the next signer, trigger the chain of signatures.

For the method 2) there is no need to inform every signer because the final envelope will be created by a single user who should know which signers are valid.

@mr0grog: Hmmm, so I think what I was getting at was a scenario like this:

1. Account A requires signatures from accounts B and C to perform a transaction.

2. C's keys are compromised, so C goes through the procedure listed here to create account D, which will have sole signing authority over C (so the old C keys no longer work).

3. At this point, what happens for A? Do the old C keys work to sign a transaction by A? (I would think not, but want to make sure.) Can D sign a transaction from A by virtue of being the only signer on C? Or does A need to remove C as a signer and add D?

In other words, is it **safe** for A to do nothing? (So long as the original owner of C was responding ASAP, of course.) Or does A need to make sure to respond as well by removing C? (They probably should, but the question is whether they have to.)